### ISSUE TITLE
scipy.sparse.linalg.eigs gives incorrect largest magnitude eigenvalue

### ISSUE THREAD

**Issue by mhauru** [scipy/scipy#8307]
For the matrix [trouble_matrix.zip](https://github.com/scipy/scipy/files/1641045/trouble_matrix.zip), eigs gives the wrong dominant eigenvalue. The result depends on the initial `v0`. For fun, I added a 6 line piece at the end that finds the correct dominant eigenvalue with a dead simple power iteration.

```
import sys
import pickle
import numpy as np
import scipy as sp
import scipy.sparse.linalg as spsla

print(sp.__version__, np.__version__, sys.version_info)

with open("trouble_matrix.p", "rb") as f:
    M = pickle.load(f)

print(type(M), M.shape)

nev = 1
d = M.shape[0]

# Exact, full eig.
S, U = np.linalg.eig(M)
S = S[np.argsort(-np.abs(S))]
print(S[:nev])

# Sparse eig, with different initial guesses.
v0 = np.random.rand(d)
S, U = spsla.eigs(M, k=nev, v0=v0)
print(S)
S, U = spsla.eigs(M, k=nev, v0=v0)
print(S)
v0 = np.random.rand(d)
S, U = spsla.eigs(M, k=nev, v0=v0)
print(S)

# Finding the dominant eigenvalue with a simple power iteration.
v = np.random.rand(d)
for _ in range(100):
    v /= np.linalg.norm(v)
    old_v = v
    v = np.dot(M, v)
print(np.average(v/old_v))
```

Output:
```
1.0.0 1.14.0 sys.version_info(major=3, minor=6, micro=3, releaselevel='final', serial=0)
<class 'numpy.ndarray'> (1296, 1296)
[0.65419337-8.24548726e-17j]
[0.84454248+22.62373266j]
[0.84454248+22.62373266j]
[23.2905353-0.77272713j]
(0.6541933722683013+1.0231927548628299e-16j)
```

One would expect all five numbers above to be the same (to precision). If `M` is picked to be a random matrix instead, this indeed happens. Asking for more than one dominant eigenvalue goes equally wrong, I'm just choosing to showcase the simplest possible thing. I tried increasing `ncv` and `maxiter`, but to no help.

There shouldn't be anything weird about the matrix. The dominant eigenvalue is not degenerate (not shown in the output below). It's not just this one matrix for which this happens, I found a bunch and just picked this as an example. 

I found a similar sounding issue for `which="LR"`, but I don't know if I should expect that to be related: https://github.com/scipy/scipy/issues/6827

I assume I'm using the ARPACK version bundled with scipy 1.0.0, which I think is supposed to be 3.3.0. I can check if you let me know how.

**Comment by pv** [comment]
Seems general ARPACK issue, since similar behavior occurs in Octave:
```
$ python
 >>> import pickle, scipy.io
 >>> with open('trouble_matrix.p', 'rb') as f:
...     M = pickle.load(f)
 >>> scipy.io.savemat('trouble.mat', dict(M=M))

$ octave
octave:1> load trouble.mat
octave:2> max(abs(eig(M)))
ans =  0.65419
octave:3> abs(eigs(M,1))
ans =  12.438
octave:4> abs(eigs(M,1))
ans =  19.764
```
Arpack bug tracker is at: https://github.com/opencollab/arpack-ng/issues
I guess they'd prefer test matrix in non-pickle format.

**Comment by pv** [comment]
The which="LR" issue is probably separate, because it concerns different 
operating mode.

**Comment by mhauru** [comment]
Thanks for the Octave check. Filed this with ARPACK: https://github.com/opencollab/arpack-ng/issues/79 Up to you whether you want to close this one now.

**Comment by denis-bz** [comment]
This matrix is incredibly ill-conditioned, numpy eigvals and svd --

    (1296,) quantiles [ 0  10  25         50    75  90 100]
    |np eigvals|      [1e-14 8e-11 3e-9   3e-7  2e-5 0.00043 0.65]
    np svd            [8e-24 1e-16 2e-12  2e-7  0.13 5.3e+03 1.3e+11]
    |M|_ij            [2e-47 1e-25 1e-19  1e-11 6e-5 0.53 9.4e+10]

So I'd say, forget it, close.

(Are there *real* testcases for arpack ?  Plottable ones would be nice,
smaller than say 10k x 10k so one could compare with numpy would be nice.
https://sparse.tamu.edu has lots of matrices, but don't know which are 
numerically singular, have closely-spaced eigenvalues ...)

(What's the current size limit for dense eigvals on a big fast public machine ?)

**Comment by mhauru** [comment]
Even if it is ill-conditioned, note that a simple power iteration still works on it.

Anyway, the ARPACK folks seem to have kindly fixed this: https://github.com/opencollab/arpack-ng/issues/79
The fix is included in ARPACK 3.7.0, SciPy just hasn't upgraded to it yet.

**Comment by ilayn** [comment]
SciPy is now on Arpack 3.9.0 hence closing this one.
